solarforecastarbiter/reference_forecasts/persistence.py

```diff
@@ -587,7 +590,10 @@ def persistence_probabilistic_timeofday(observation, data_start, data_end,
 
     # observation data resampled to match forecast sampling
     obs = load_data(observation, data_start, data_end)
-    obs = obs.resample(interval_length, closed=closed).mean()
+    # unlike other functions in this module, we do need label=closed
+    # here because we will match the resampled obs time of day to
+    # the desired forecast time of day.
+    obs = obs.resample(interval_length, closed=closed, label=closed).mean()
```

Comment:
I think the closed parameter value here should be based off of the interval label of the observation rather than the interval label of the desired forecast. I think the value currently passed as label is correct though.

For example, if we have an observation with a beginning labeled 10minute interval and we want to resample to an hour interval with ending labels: 
```
# hour of 0, hour of 1, 5 minutes of 2
idx = pd.date_range('2021-01-01T00:00Z', '2021-01-01T02:00Z', freq='10T')
data = [0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1,2]
ser = pd.Series(data, index=idx)

>>> ser.resample('1H', closed="right", label="right").mean()
2021-07-29 00:00:00    0.000000
2021-07-29 01:00:00    0.083333
2021-07-29 02:00:00    1.083333
2021-07-29 03:00:00    2.000000
Freq: H, dtype: float64
>>> ser.resample('1H', closed="left", label="right").mean()
2021-07-29 01:00:00    0
2021-07-29 02:00:00    1
2021-07-29 03:00:00    2
Freq: H, dtype: int64
```
It seems like if closed isn't set to the labeling convention of the observation, there's a bleed over of one observation interval when resampling, or nothing happens at all if the forecast and observation have the same interval length. 
```
>>> data = [1,2,3]                                                                                    
>>> idx = pd.date_range('2021-01-01T00:00Z', '2021-01-01T02:00Z', freq='1H')
>>> ser = pd.Series(data, idx)
# closed and mapping same has no effect
>>> ser.resample("1h", closed="right", label="right").mean()
2021-01-01 00:00:00+00:00    1
2021-01-01 01:00:00+00:00    2
2021-01-01 02:00:00+00:00    3
Freq: H, dtype: int64
# closed=left for beginning observation and label=right for ending forecast works as expected
>>> ser.resample("1h", closed="left", label="right").mean()
2021-01-01 01:00:00+00:00    1
2021-01-01 02:00:00+00:00    2
2021-01-01 03:00:00+00:00    3
Freq: H, dtype: int64
```

Reply:
You're right. Very good catch. I reviewed the other persistence functions and they are fine.